### ISSUE TITLE
Accuracy Metrics

### ISSUE THREAD

**Issue by jianghc93** [inspirehep/magpie#103]
Hi I see that you are using the 'accuracy' metric in the neural networks.

When dealing with multilabel classification does it mean that if any one of the tags are correct, the prediction is treated as correct? i.e. Is the following case considered a correct prediction:

Truth label: [0, 1, 2]
Predicted label: [2, 3, 4, 5]

Would it be better for a custom metric like F1 score?

**Comment by jstypka** [comment]
We use Keras' metrics, so in this example it will not be treated as fully correct - only as 1/6 correct, because just one out of 6 labels was correctly predicted. [Here's the code](https://github.com/fchollet/keras/blob/master/keras/metrics.py#L20-L21) for that.

But you're right, that there's no good reason for using accuracy in this case. You can compute custom metrics using the callbacks in the `train` function, but Magpie should probably provide an additional metric on top of the loss. F1 score sounds like a good suggestion.

**Comment by jianghc93** [comment]
Do you mean 1/4? Since only 4 labels are predicted. 

If I'm not wrong, the binary_accuracy takes in a one hot vector for the truth labels. So in the example it would be: 

Truth label:  [1, 1, 1, 0, 0, 0]
Predicted label after rounding: [0, 0, 1, 1, 1, 1]

Since only one is correct. K.mean gives us number of correct / number of classes = 1/6 (Is this what you were implying at the start?)

This is kind of weird because even if the prediction exactly matches the truth labels we would only be getting 3/6?

**Comment by jianghc93** [comment]
Oh wait my mistake. I forgot that the 0's are also considered as correct predictions. So exact match will give us 6/6.

Thanks for the help
